### ISSUE TITLE
[AVM Module Issue]: Unsupported attribute when declaring 'shutdown_schedules'

### ISSUE THREAD

**Issue by ebrookman** [Azure/terraform-azurerm-avm-res-compute-virtualmachine#82]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### (Optional) Module Version

0.13.0

### Description

when using the example code in the README.md file, I get an attribute unexpected. 

``` hcl
module "primary_testvm" {
  count   = 1
  source  = "Azure/avm-res-compute-virtualmachine/azurerm"
  version = "0.13.0"

  admin_username                         = "admin"
  enable_telemetry                       = false
  resource_group_name                    = data.terraform_remote_state.mgmt.outputs.primary_bld_odb_rg
  location                               = var.primary_region
  name                                   = upper(format("testvm%s%s", (count.index + 1), var.primary_region_short_name))
  virtualmachine_sku_size                = "Standard_D8s_v5"
  virtualmachine_os_type                 = "Linux"
  zone                                   = 1
  admin_credential_key_vault_resource_id = data.terraform_remote_state.mgmt.outputs.primary_keyvault["resource"]["id"]
  enable_automatic_updates               = false
  encryption_at_host_enabled             = false
  disable_password_authentication        = false

  shutdown_schedules = {
    test_schedule = {
      daily_recurrence_time = "1700"
      enabled               = true
      timezone              = "Pacific Standard Time"
      notification_settings = {
        enabled         = true
        email           = "example@example.com;example2@example.com"
        time_in_minutes = "15"
        webhook_url     = "https://example-webhook-url.example.com"
      }
    }
  }

```

**Comment by jchancellor-ms** [comment]
@ebrookman - thanks for reporting this.  I'll try to reproduce and diagnose the issue and provide an update.

**Comment by jchancellor-ms** [comment]
@ebrookman - I ran the following test in my lab subscription with success.  Would you be able to provide any additional context on any error details about which attribute is causing issues?

``` hcl
module "primary_testvm" {
  count   = 1
  source  = "Azure/avm-res-compute-virtualmachine/azurerm"
  version = "0.13.0"

  admin_username                         = "testuser"
  enable_telemetry                       = false
  resource_group_name                    = azurerm_resource_group.this_rg.name
  location                               = azurerm_resource_group.this_rg.location
  name                                   = upper(format("testvm%s%s", (count.index + 1), "test"))
  virtualmachine_sku_size                = module.get_valid_sku_for_deployment_region.sku
  virtualmachine_os_type                 = "Linux"
  zone                                   = 1
  admin_credential_key_vault_resource_id = module.avm_res_keyvault_vault.resource.id
  enable_automatic_updates               = false
  encryption_at_host_enabled             = false
  disable_password_authentication        = false

  shutdown_schedules = {
    test_schedule = {
      daily_recurrence_time = "1700"
      enabled               = true
      timezone              = "Pacific Standard Time"
      notification_settings = {
        enabled         = true
        email           = "example@example.com;example2@example.com"
        time_in_minutes = "15"
        webhook_url     = "https://example-webhook-url.example.com"
      }
    }
  }

  network_interfaces = {
    network_interface_1 = {
      name = "${module.naming.network_interface.name_unique}-test"
      ip_configurations = {
        ip_configuration_1 = {
          name                          = "${module.naming.network_interface.name_unique}-test-ipconfig1"
          private_ip_subnet_resource_id = azurerm_subnet.this_subnet_1.id
        }
      }
    }
  }

  os_disk = {
    caching              = "ReadWrite"
    storage_account_type = "StandardSSD_LRS"
  }

  source_image_reference = {
    publisher = "Canonical"
    offer     = "0001-com-ubuntu-server-focal"
    sku       = "20_04-lts-gen2"
    version   = "latest"
  }

  tags = local.tags

  depends_on = [
    module.avm_res_keyvault_vault
  ]
}
```

![image](https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/assets/84210452/6a3a3096-c523-41bd-8318-4006aafcb1f0)

**Comment by ebrookman** [comment]
When trying to deploy I was getting an error that said the variable wasn’t expected even after doing the initial plan.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jchancellor-ms** [comment]
When you have a moment can you please provide me a list of the provider versions you are using? (or just a screenshot of your terraform init output if it doesn't contain anything sensitive)

**Comment by ebrookman** [comment]
<img width="476" alt="image" src="https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/assets/152096962/7b3b724c-a90f-43e7-b567-b8240385ea80">

**Comment by ebrookman** [comment]
<img width="641" alt="image" src="https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/assets/152096962/00938646-7e1f-470c-9ee7-e058b6f73ec6">

**Comment by jchancellor-ms** [comment]
Sorry - Can you please provide the output of `terraform init`?  I'd like to validate the module versions that are being used.  
![image](https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/assets/84210452/10a43c5a-598c-49db-b851-4426f6a607b5)

**Comment by jchancellor-ms** [comment]
ok - let me pin on your provider versions and see if I get the same errors. - thanks

**Comment by jchancellor-ms** [comment]
No issues with the versions.  I did some quick research on the shutdown feature and it looks like it uses the devtestlabs api. Can you check to see if you have the `Microsoft.DevTestLab` resource provider enabled on your subscription? 
![image](https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/assets/84210452/773b3a68-b72e-4865-88df-2b293177b7b9)

**Comment by ebrookman** [comment]
Just got it confirmed that this is enabled on my subs.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by jchancellor-ms** [comment]
I'm trying to eliminate a few more possibilities.  Can you please let me know which version of terraform you're using, and which region you're trying to deploy into?  Also, I've released version 0.14 with some new unrelated fixes this morning. Could you please try updating to the 0.14 release and re-running a terraform init -upgrade to eliminate any possible provider download issues?

**Comment by jchancellor-ms** [comment]
Closing this issue as this should be corrected by forcing the provider versions to a current version.  Release version 0.15 now requires AzureRM 3.108 which should not include this issue.  Please let me know if you're still encountering the issue and we can re-open.
